Fixing issues with Qt5.14 on Windows and Plugins directory.

imikejackson opened the pull request:
Qt 5.14 allows relocatable Qt installations and so now we either need
to copy in all of the plugins from the Qt installation or just write
out a qt.conf file that points to the proper location for the Qt Plugins
directory. We chose the second option.

Signed-off-by: Michael Jackson <mike.jackson@bluequartz.net>

JDuffeyBQ commented on ExtLib/Qt5Support.cmake:
```diff
@@ -151,36 +118,24 @@ function(AddQt5SupportLibraryCopyInstallRules)
+      foreach(INT_DIR ${OUTPUT_DIRS})
+        # message(STATUS "Copy Rule for Qt Support library ${qtlib}${SUFFIX}.dll")
+        if(NOT TARGET ZZ_${qtlib}-${INT_DIR}-Copy)
+          set(SUFFIX ${P_DEBUG_SUFFIX})
```
This will apply the debug suffix to all configurations.

JDuffeyBQ commented on ExtLib/Qt5Support.cmake:
```diff
@@ -204,7 +159,71 @@ function(AddQt5SupportLibraryCopyInstallRules)
+
+  cmake_parse_arguments(P "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN} )
+
+  set(SUPPORT_LIB_OPTION 1)
+  if(MSVC_IDE)
+    set(SUPPORT_LIB_OPTION 0)
+  elseif(APPLE) # Apple systems do NOT need this so just skip this entirely
+    set(SUPPORT_LIB_OPTION 2)
+  elseif(UNIX AND NOT MSVC)
+    set(SUPPORT_LIB_OPTION 3)
+  endif()
```
We could move these conditions down to their respective if statements and have an early return for APPLE.

JDuffeyBQ replied:
Also since only the file itself and the plugin path change from case to case, only those need to be inside the if statements.

JDuffeyBQ commented on ExtLib/Qt5Support.cmake:
```diff
@@ -204,7 +159,71 @@ function(AddQt5SupportLibraryCopyInstallRules)
+      FILE(WRITE ${qt_conf_file} "[Paths]\nPlugins = ${P_QT_PLUGINS_DIR}\n")
+      FILE(APPEND ${qt_conf_file} "Prefix = .\n")
+      FILE(APPEND ${qt_conf_file} "LibraryExecutables = .\n")
+      FILE(APPEND ${qt_conf_file} "Data = .\n")
```
```suggestion
      file(WRITE ${qt_conf_file} "[Paths]\nPlugins = ${P_QT_PLUGINS_DIR}\n")
      file(APPEND ${qt_conf_file} "Prefix = .\n")
      file(APPEND ${qt_conf_file} "LibraryExecutables = .\n")
      file(APPEND ${qt_conf_file} "Data = .\n")
```

JDuffeyBQ commented on ExtLib/Qt5Support.cmake:
```diff
@@ -204,7 +159,71 @@ function(AddQt5SupportLibraryCopyInstallRules)
+    FILE(WRITE ${qt_conf_file} "[Paths]\nPlugins = ${P_QT_PLUGINS_DIR}\n")
+    FILE(APPEND ${qt_conf_file} "Prefix = .\n")
+    FILE(APPEND ${qt_conf_file} "LibraryExecutables = .\n")
+    FILE(APPEND ${qt_conf_file} "Data = .\n")
```
```suggestion
    file(WRITE ${qt_conf_file} "[Paths]\nPlugins = ${P_QT_PLUGINS_DIR}\n")
    file(APPEND ${qt_conf_file} "Prefix = .\n")
    file(APPEND ${qt_conf_file} "LibraryExecutables = .\n")
    file(APPEND ${qt_conf_file} "Data = .\n")
```

JDuffeyBQ commented on ExtLib/Qt5Support.cmake:
```diff
@@ -204,7 +159,71 @@ function(AddQt5SupportLibraryCopyInstallRules)
+    FILE(WRITE ${qt_conf_file} "[Paths]\nPlugins = ${QTPLUGINS_DIR}Plugins\n")
+    FILE(APPEND ${qt_conf_file} "Prefix = .\n")
+    FILE(APPEND ${qt_conf_file} "LibraryExecutables = .\n")
+    FILE(APPEND ${qt_conf_file} "Data = .\n")
```
```suggestion
    file(WRITE ${qt_conf_file} "[Paths]\nPlugins = ${QTPLUGINS_DIR}Plugins\n")
    file(APPEND ${qt_conf_file} "Prefix = .\n")
    file(APPEND ${qt_conf_file} "LibraryExecutables = .\n")
    file(APPEND ${qt_conf_file} "Data = .\n")
```

JDuffeyBQ reviewed (changes requested):
Some of the other changes should probably also be documented.

imikejackson commented:
I refactored more of the logic statements, added early returns for APPLE systems and ensured that Debug is not applied to all configurations on Visual Studio IDE builds

JDuffeyBQ commented on ExtLib/Qt5Support.cmake:
```diff
@@ -442,7 +395,7 @@ macro(CMP_AddQt5Support Qt5Components ProjectBinaryDir VarPrefix)
   endif()
 
-
+  AddQtConfTargets(QT_PLUGINS_DIR ${QM_QT_INSTALL_PLUGINS})
```
```suggestion
  if(NOT DREAM3D_ANACONDA)
    AddQtConfTargets(QT_PLUGINS_DIR ${QM_QT_INSTALL_PLUGINS})
  endif()
```

I just tested with our conda build, and this looked to be the only thing that needed to be fixed.

imikejackson commented:
Have we tested this against Qt 5.12 on Windows yet?
